ports/samd/machine_spi.c

```diff
@@ -132,14 +146,16 @@ static void machine_spi_init(mp_obj_base_t *self_in, size_t n_args, const mp_obj
     if (args[ARG_mosi].u_obj != mp_const_none) {
         self->mosi = mp_hal_get_pin_obj(args[ARG_mosi].u_obj);
     }
-    if (args[ARG_miso].u_obj != mp_const_none) {
+    if (args[ARG_miso].u_int != -1 && args[ARG_miso].u_obj != mp_const_none) {
         self->miso = mp_hal_get_pin_obj(args[ARG_miso].u_obj);
     }
-
     // Initialise the SPI peripheral if any arguments given, or it was not initialised previously.
     if (n_args > 0 || kw_args->used > 0 || self->new) {
         self->new = false;
 
+        if (self->sck == 0xff || self->mosi == 0xff) {
+            mp_raise_ValueError(MP_ERROR_TEXT("mosi or sck missing"));
```

Comment:
If you want to reduce code size, I suggest "missing sck/mosi".

Reply:
Reducing the code size is always good, even by a few bytes.